sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -269,8 +269,10 @@ static bool isDeviceBinaryTypeSupported(const context &C,
       return false;
   }
 
+  ContextImplPtr ContextImpl = getSyclObjImpl(C);
+
   // OpenCL 2.1 and greater require clCreateProgramWithIL
-  if (pi::useBackend(pi::SYCL_BE_PI_OPENCL) &&
+  if (ContextImpl->getPlugin().isBackendType(Backend::SYCL_BE_PI_OPENCL) &&
```

Comment:
This could be instead:

```cpp
ContextImpl->getPlugin()->getBackend() == Backend::SYCL_BE_PI_OPENCL
```

Reply:
as above, getBackendType ? if you prefer, i will do that :)

Reply:
I would say `getBackend` is enough, since the return type encodes what the type is (a Backend)